src/main/java/net/atos/zac/app/util/EnkelvoudigInformatieObjectStatusEnumReader.java

```diff
@@ -0,0 +1,53 @@
+package net.atos.zac.app.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.ext.MessageBodyReader;
+import jakarta.ws.rs.ext.Provider;
+
+import org.apache.commons.lang3.StringUtils;
+
+import net.atos.client.zgw.drc.model.generated.EnkelvoudigInformatieObject;
+
+@Provider
+@Produces(MediaType.TEXT_PLAIN)
+public class EnkelvoudigInformatieObjectStatusEnumReader implements MessageBodyReader<EnkelvoudigInformatieObject.StatusEnum> {
+
+    private static final String CLASS_PREFIX = EnkelvoudigInformatieObject.class.toString().replace("class ", "") + ".StatusEnum.";
+
+    @Override
+    public boolean isReadable(Class<?> aClass, Type type, Annotation[] annotations, MediaType mediaType) {
+        return type == EnkelvoudigInformatieObject.StatusEnum.class;
+    }
+
+    @Override
+    public EnkelvoudigInformatieObject.StatusEnum readFrom(
+            Class<EnkelvoudigInformatieObject.StatusEnum> aClass,
+            Type type,
+            Annotation[] annotations,
+            MediaType mediaType,
+            MultivaluedMap<String, String> multivaluedMap,
+            InputStream inputStream
+    ) throws IOException, WebApplicationException {
+        if (inputStream == null) {
+            return null;
+        }
+
+        byte[] localDateBytes = inputStream.readAllBytes();
+        if (localDateBytes.length == 0) {
+            return null;
+        }
+
+        String enumValueAsString = new String(localDateBytes);
+        enumValueAsString = StringUtils.removeStart(enumValueAsString, CLASS_PREFIX);
+
+        return EnkelvoudigInformatieObject.StatusEnum.valueOf(enumValueAsString.toUpperCase());
```

Comment:
Could/should this not be `.fromValue(enumValueAsString)`? I did that `toUpperCase()` trick in other places in the code when we switched from the old model classes to the generated model classes but in hindsight I think those should have been `fromValue` as well.

Reply:
Ah, never mind. This is not a (custom) enum class so this does not apply here at all..